<a href="https://colab.research.google.com/github/zyf2021/-/blob/main/%D0%AF%D0%BD%D0%B4%D0%B5%D0%BA%D1%81%D0%9A%D0%BE%D0%BD%D1%82%D0%B5%D1%81%D1%82_%D0%9F%D1%80%D0%B8%D0%B2%D0%B5%D0%B4%D0%B5%D0%BD%D0%B8%D0%B5_%D0%B4%D0%B0%D0%BD%D0%BD%D1%8B%D1%85_%D0%BA_%D1%86%D0%B5%D0%BB%D0%B5%D0%B2%D0%BE%D0%BC%D1%83_%D0%B2%D0%B8%D0%B4%D1%83.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
Vasya = 'Vasya'
print('Vasya', 'Masha', sep = '#')

Vasya#Masha


In [13]:
# подключение гугл диска, на котором хранится zip-архив с датасетом
from google.colab import drive
drive.mount('/content/drive/')

Drive already mounted at /content/drive/; to attempt to forcibly remount, call drive.mount("/content/drive/", force_remount=True).


#### Что нас волнует?

В предыдущих главах мы работали с данными без их предварительной очистки.

В этой и следующих задачах мы используем набор данных с фильмами и сериалами компании Disney. CSV файл доступен под названием 'disney_title.csv'. Ознакомиться более подробно с датасетом можно по следующей ссылке: https://www.kaggle.com/datasets/shivamb/disney-movies-and-tv-shows.

В этой задаче выведите информацию о датасете, используя функцию .info()

In [14]:
import pandas as pd

data = pd.read_csv('/content/drive/MyDrive/disney_plus_titles.csv')
print(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1450 entries, 0 to 1449
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       1450 non-null   object
 1   type          1450 non-null   object
 2   title         1450 non-null   object
 3   director      977 non-null    object
 4   cast          1260 non-null   object
 5   country       1231 non-null   object
 6   date_added    1447 non-null   object
 7   release_year  1450 non-null   int64 
 8   rating        1447 non-null   object
 9   duration      1450 non-null   object
 10  listed_in     1450 non-null   object
 11  description   1450 non-null   object
dtypes: int64(1), object(11)
memory usage: 136.1+ KB
None


#### Какая сегодня дата?
В предыдущем задании мы обнаружили, что переменная Date из датасета по фильмам и сериалам компании Disney имеет тип object. В этом задании мы поменяем тип переменной с object на datetime с помощью функции pd.to_datetime(). В выводе распечатайте тип столбца ['Date'] с помощью .dtype

CSV файл доступен под названием 'disney_title.csv'.


Это решение почему-то не принято из-за времени

In [15]:
#data['date_added'] = pd.to_datetime(data['date_added'])
data['date_added'] = data['date_added'].astype('datetime64[ns]')
data['date_added'].dtype

dtype('<M8[ns]')

#### А что выходило за год?

В предыдущем задании мы изменили тип переменной, содержащей дату, с объекта на значение даты и времени. В этом задании мы отфильтруем датасет с использованием переменной. Вначале еще раз измените тип переменной Date на datetime64 и отфильтруйте все фильмы и сериалы, которые выходили с 1 января 2020 года по 1 января 2021 года. В результате выведите переменную с названиями десяти первых фильмов после фильтрации, с помощью функции head().

In [16]:
data['date_added'] = data['date_added'].astype('datetime64[ns]')
data_filtered = data.loc[(data['date_added'] >= '01-01-2020') & (data['date_added'] <= '01-01-2021')]
print(data_filtered.head())

    show_id   type                           title  \
320    s321  Movie                      700 Sharks   
321    s322  Movie                 Big Sharks Rule   
322    s323  Movie  Dr. Seuss' Horton Hears a Who!   
323    s324  Movie           Great Shark Chow Down   
324    s325  Movie                 Mega Hammerhead   

                         director  \
320                  Luc Marescot   
321                           NaN   
322  Jimmy Hayward, Steve Martino   
323                 Mark Woodward   
324    Brandon Hill, Marc Ostrick   

                                                  cast        country  \
320                                        Paul Bandey         France   
321                                       Lance Lewman            NaN   
322  Jim Carrey, Steve Carell, Will Arnett, Seth Ro...  United States   
323                                       Dan Nachtrab            NaN   
324                                  Neil Hammerschlag  United States   

    date_adde

#### Удаление

В этом задании мы продолжаем работать с CSV файлом под названием 'disney_title.csv'. Вам необходимо удалить столбец из датасета 'release_year' и вывести название оставшихся столбцов с помощью функции list().

In [17]:
data = data.drop(labels = 'release_year', axis = 1)
list(data)

['show_id',
 'type',
 'title',
 'director',
 'cast',
 'country',
 'date_added',
 'rating',
 'duration',
 'listed_in',
 'description']

#### А можно переименовать?

Умение переименовывать датасеты помогает при работе с большими объемами данных. В этом задании вам необходимо поменять название переменных из 'disney_title.csv' на такие же названия написанные с большой буквы. В результате выведите названия стобцов с помощью функции list().

In [19]:
cap_columns = [col.capitalize() for col in list(data)]
data.columns = cap_columns
list(data)

['Show_id',
 'Type',
 'Title',
 'Director',
 'Cast',
 'Country',
 'Date_added',
 'Rating',
 'Duration',
 'Listed_in',
 'Description']

#### А это что за покемон?

В этом задании мы продолжаем работать с CSV файлом доступным под названием 'disney_title.csv'.

В этом задании вам необходимо заменить "&" на "," в столбце "listed_in", сохранить результат в новую переменную вывести последнии пять значений в новой переменной с помощью функции 'tail()'.

In [21]:
data['Listed_in'] = data['Listed_in'].str.replace('&', ',')
data.tail()

Show_id   Type                                           Title  \
1445   s1446  Movie                        X-Men Origins: Wolverine   
1446   s1447  Movie  Night at the Museum: Battle of the Smithsonian   
1447   s1448  Movie                                 Eddie the Eagle   
1448   s1449  Movie                            Bend It Like Beckham   
1449   s1450  Movie           Captain Sparky vs. The Flying Saucers   

             Director                                               Cast  \
1445       Gavin Hood  Hugh Jackman, Liev Schreiber, Danny Huston, wi...   
1446       Shawn Levy  Ben Stiller, Amy Adams, Owen Wilson, Hank Azar...   
1447  Dexter Fletcher  Tom Costello, Jo Hartley, Keith Allen, Dickon ...   
1448  Gurinder Chadha  Parminder Nagra, Keira Knightley, Jonathan Rhy...   
1449      Mark Waring                                      Charlie Tahan   

                                     Country Date_added Rating Duration  \
1445           United States, United Kingdom 2021-06-04  PG-13  108 min   
1446                   United States, Canada 2021-04-02     PG  106 min   
1447  United Kingdom, Germany, United States 2020-12-18  PG-13  107 min   
1448  United Kingdom, Germany, United States 2020-09-18  PG-13  112 min   
1449                           United States 2020-04-01   TV-G    2 min   

                                          Listed_in  \
1445      Action-Adventure, Family, Science Fiction   
1446               Action-Adventure, Comedy, Family   
1447                    Biographical, Comedy, Drama   
1448                   Buddy, Comedy, Coming of Age   
1449  Action-Adventure, Animals , Nature, Animation   

                                            Description  
1445  Wolverine unites with legendary X-Men to fight...  
1446  Larry Daley returns to rescue some old friends...  
1447  True story of Eddie Edwards, a British ski-jum...  
1448  Despite the wishes of their traditional famili...  
1449         View one of Sparky's favorite home movies.

### Сколько пропусков?

В этой задаче требуется вывести количество пропущенных значений по каждому стобцу.

In [23]:
data.isnull().sum()

Show_id          0
Type             0
Title            0
Director       473
Cast           190
Country        219
Date_added       3
Rating           3
Duration         0
Listed_in        0
Description      0
dtype: int64

#### Удаление?

CSV-файл доступен под названием `'disney_title.csv'`. По результатам седьмого задания мы выяснили, что небольшое количество пропущенных значений (по три) находятся в столбцах `'rating'` и `'Date'`. Воспользуйтесь функцией `'dropna()'` для удаление пропущенных значений во всех столбцах. В результате выведите количество пропущенных переменных по каждому столбцу в вашем датасете.

In [24]:
copy_data = data.dropna(how = 'any')
copy_data.isnull().sum()

Show_id        0
Type           0
Title          0
Director       0
Cast           0
Country        0
Date_added     0
Rating         0
Duration       0
Listed_in      0
Description    0
dtype: int64

#### Какой процент пропусков?

В предыдущем задании мы смотрели на пропущенные значения без анализа их количества относительно количество заполненных значений во всем датасете. В этом задании мы продолжим работать с CSV файлом под названием 'disney_title.csv'. Выведите процент пропущенных значений относительного всех значений. Результат округлите до сотых с помощью функции "round()".

In [25]:
data.isnull().mean()*100

Show_id         0.000000
Type            0.000000
Title           0.000000
Director       32.620690
Cast           13.103448
Country        15.103448
Date_added      0.206897
Rating          0.206897
Duration        0.000000
Listed_in       0.000000
Description     0.000000
dtype: float64

#### Замена пропусков

Из предыдущего задания мы выяснили, что наибольшие проценты пропущенных значений в CSV файле 'disney_title.csv' находятся в стобцах director (32.62%), cast (13.10%) и country (15.10%). В этом задании заменить пропущенные значения в стобце 'country' на 'Country not specified' с помощью функции 'fillna'. Выведите первые пять первых значений переменной 'country'.

In [29]:
data['Director'].fillna('Country not specified', inplace = True)
data['Cast'].fillna('Country not specified', inplace = True)
data['Country'].fillna('Country not specified', inplace = True)
data['Country'].head()

0    Country not specified
1    Country not specified
2            United States
3    Country not specified
4    Country not specified
Name: Country, dtype: object